### DISCUSSION TITLE
Where is the source code when installation is done using DigitalOcean Marketplace

### DISCUSSION THREAD

**Discussion by cptalpdeniz** [getredash/redash#6587] [Q&A] [answered]
Hello,

I setup a DigitalOcean droplet then installed Redash using DigitalOcean Marketplace. As far as I see, this uses docker. The issue is I need to make changes to the files (for example change logo) and few other things but I never used docker and I cannot find the files in `/var/www/` as well. I was only able to find `/opt/redash/` but then this is also pretty empty. Where are the source files and how do I use docker with it?

**Comment by cptalpdeniz** [comment]
I mean I don't understand this at all, I just need to change ONE file, server is running so clearly this file is SOMEWHERE in the somewhere but I cannot find it, I can't even find the file using `find` command run on `/`, most stupid thing I have ever seen... Literally wasted 8 hours trying to find where the files are just because of this docker thing. Is there ANY HELP available?

**Reply by masayuki038** [reply]
Hi @cptalpdeniz, sorry for the delay in following up.

I haven't used Digital Ocean... If you are using Docker, files are in:

### Client
/app/client/dist in the docker container

### Server
/app in the docker container

For example, my local env:
```
[masayuki@localhost redash]$ docker exec -it redash-server-1 bash # login to redash container
redash@96e5f3c4e3eb:/app$ ls -l /app #server
total 1276
-rw-r--r--.    1 redash redash  93731 Oct 24 11:24 CHANGELOG.md
-rw-r--r--.    1 redash redash   5857 Oct 24 11:24 CONTRIBUTING.md
-rw-r--r--.    1 redash redash   3245 Oct 24 11:24 Dockerfile
-rw-r--r--.    1 redash redash   1320 Oct 24 11:24 LICENSE
-rw-r--r--.    1 redash redash    195 Oct 24 11:24 LICENSE.borders
-rw-r--r--.    1 redash redash   1445 Oct 24 11:24 Makefile
-rw-r--r--.    1 redash redash   4414 Oct 24 11:24 README.md
-rw-r--r--.    1 redash redash    429 Oct 24 11:24 SECURITY.md
drwxr-xr-x.    2 redash redash     90 Oct 24 11:24 bin
drwxr-xr-x.    5 redash redash    166 Oct 24 12:03 client
-rw-r--r--.    1 redash redash    128 Oct 24 11:24 codecov.yml
-rw-r--r--.    1 redash redash    624 Oct 24 11:24 cypress.config.js
-rw-r--r--.    1 redash redash   1932 Nov 10 12:35 docker-compose.yml
-rwxr-xr-x.    1 redash redash    126 Oct 24 11:24 manage.py
drwxr-xr-x.    4 redash redash    133 Oct 24 12:26 migrations
-rw-r--r--.    1 redash redash   1344 Oct 24 11:24 netlify.toml
drwxr-xr-x. 1418 redash redash  40960 Oct 24 12:02 node_modules
-rw-r--r--.    1 redash redash   6667 Nov  6 11:54 package.json
-rw-r--r--.    1 redash redash 356261 Nov  8 13:19 poetry.lock
-rw-r--r--.    1 redash redash   3769 Nov  8 13:19 pyproject.toml
-rw-r--r--.    1 redash redash    140 Oct 24 11:24 pytest.ini
drwxr-xr-x.   15 redash redash   4096 Nov  6 11:54 redash
drwxr-xr-x.    2 redash redash     23 Oct 24 11:24 scripts
drwxr-xr-x.    2 redash redash     23 Oct 24 11:24 setup
drwxr-xr-x.   10 redash redash   4096 Nov  6 12:52 tests
drwxr-xr-x.    6 redash redash   4096 Oct 24 11:35 viz-lib
-rw-r--r--.    1 redash redash   7801 Oct 24 11:24 webpack.config.js
-rw-r--r--.    1 redash redash    760 Oct 24 11:24 worker.conf
-rw-r--r--.    1 redash redash 689976 Nov  6 11:54 yarn.lock
redash@96e5f3c4e3eb:/app$ ls -l /app/client/dist #client
total 26280
-rw-r--r--. 1 redash redash   999294 Oct 24 12:03 app.3df198261ff39a42a5fb.css
-rw-r--r--. 1 redash redash  1373038 Oct 24 12:03 app.3df198261ff39a42a5fb.css.map
-rw-r--r--. 1 redash redash   630183 Oct 24 12:03 app.3df198261ff39a42a5fb.js
-rw-r--r--. 1 redash redash  1397532 Oct 24 12:03 app.3df198261ff39a42a5fb.js.map
-rw-r--r--. 1 redash redash    12415 Oct 24 12:03 asset-manifest.json
drwxr-xr-x. 2 redash redash      117 Oct 24 12:03 data
drwxr-xr-x. 3 redash redash     4096 Oct 24 12:03 fonts
drwxr-xr-x. 7 redash redash     4096 Oct 24 12:03 images
-rw-r--r--. 1 redash redash     1104 Oct 24 12:03 index.html
-rw-r--r--. 1 redash redash     1121 Oct 24 12:03 multi_org.html
-rw-r--r--. 1 redash redash       31 Oct 24 12:03 robots.txt
-rw-r--r--. 1 redash redash   286209 Oct 24 12:03 server.56675bd6a004ad9b79b0.css
-rw-r--r--. 1 redash redash   412768 Oct 24 12:03 server.56675bd6a004ad9b79b0.css.map
-rw-r--r--. 1 redash redash     1063 Oct 24 12:03 server.56675bd6a004ad9b79b0.js
-rw-r--r--. 1 redash redash     4776 Oct 24 12:03 server.56675bd6a004ad9b79b0.js.map
drwxr-xr-x. 2 redash redash       23 Oct 24 12:03 styles
-rw-r--r--. 1 redash redash     1535 Oct 24 12:03 unsupported.html
-rw-r--r--. 1 redash redash      155 Oct 24 12:03 unsupportedRedirect.js
-rw-r--r--. 1 redash redash   143333 Oct 24 12:03 vendors~app.9b3007ee59448f3713f7.css
-rw-r--r--. 1 redash redash   218294 Oct 24 12:03 vendors~app.9b3007ee59448f3713f7.css.map
-rw-r--r--. 1 redash redash  4131696 Oct 24 12:03 vendors~app.9b3007ee59448f3713f7.js
-rw-r--r--. 1 redash redash 17246967 Oct 24 12:03 vendors~app.9b3007ee59448f3713f7.js.map
```

**Reply by cptalpdeniz** [reply]
I do apologize for the outburst, I do desktop environment, have very simple knowledge of web and when im time limited + wasting time I just got extremely frustrated. I want to change couple of files and because things are in docker containers (which I have no knowledge of docker containers, images, mounts, volumes etc.) I'm just at a loss. I will read your comment and get back.

**Reply by cptalpdeniz** [reply]
Okay so I see that you shell into the docker container but then how to change the files? I want to setup a git repo where the files are located so that me and few groupmates can make changes

**Reply by masayuki038** [reply]
If you want to temporarily copy files, how about using `docker cp`?
https://docs.docker.com/engine/reference/commandline/cp/

**Reply by cptalpdeniz** [reply]
I'm trying bind mount so that the changes are reflected and I can setup the git on the host (which is the server) but then I'm getting `OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"/app/bin/docker-entrypoint\": stat /app/bin/docker-entrypoint: no such file or directory": unknown`

**Reply by masayuki038** [reply]
Hmm, I don't know...

If you want to change some files of Redash and deploy it continuously, I would do as follows:

- Fork redash
- Modify forked redash files
- Create a modified redash docker container
- Deploy the created container

**Reply by cptalpdeniz** [reply]
We already made changes to the current installed redash, unfortunately we cannot start over :/ I also do not know how to create a docker container

**Comment by guidopetri** [comment, marked as answer]
@cptalpdeniz sorry you've had a frustrating day. We're all volunteers here - nobody is paid to work on redash or answer questions - so please bear that in mind if you don't get a response right away.

The repo files are copied into the Redash docker container under the `/app` directory, as you can see here: https://github.com/getredash/redash/blob/master/Dockerfile#L96 this line copies the contents of `.`, i.e. the entire repo, into `/app`, changing the owner of the files to `redash`.

I'd suggest following @masayuki038 's instructions of forking the repo and modifying the files as needed, then building the container. You can follow the GHA instructions to do that, and to deploy it, you can use the `docker-compose.yml` file (pointing to your own container image) or deploy it manually with something like `docker run`. However, you'll need to learn a bit more about Docker for this, and we don't have a good documentation base that would help you. Perhaps you could open a PR or edit the wiki with some instructions once you've figured it out?
